app/lib/package/backend.dart

```diff
@@ -981,8 +981,24 @@ class PackageBackend {
     final currentDartSdk = await getDartSdkVersion();
 
     // query admin notification emails before the transaction starts
-    final uploaderEmails =
-        await _listAdminNotificationEmailsForPackage(newVersion.package, agent);
+    final existingPackage = await lookupPackage(newVersion.package);
+    List<String> uploaderEmails;
+    if (existingPackage == null) {
+      if (agent is AuthenticatedUser) {
+        uploaderEmails = [agent.email!];
+      } else {
+        throw PackageRejectedException.onlyUsersAreAllowedToUploadNewPackages();
+      }
+    } else {
+      uploaderEmails =
+          await _listAdminNotificationEmailsForPackage(existingPackage);
+    }
+    if (uploaderEmails.isEmpty) {
+      // should not happen
+      throw StateError(
+          'Package "${newVersion.package}" has no admin email to notify.');
```

Comment:
```suggestion
      // should not happen
      throw AssertionError(
          'Package "${newVersion.package}" has no admin email to notify.');
```

Reply:
nit, it's just that this isn't an object in the wrong state.

It's unreachable code, this should never happen, an invariant was violated.